onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -0,0 +1,424 @@
+    parsers_.emplace(fused_node->Name(), std::move(trt_parser));
+    engines_.emplace(fused_node->Name(), std::move(trt_engine));
+    contexts_.emplace(fused_node->Name(), std::move(trt_context));
+    input_info_[fused_node->Name()].push_back(input_indexes);
+    input_info_[fused_node->Name()].push_back(input_binding_indexes);
+    input_info_[fused_node->Name()].push_back(input_dim_sizes);
+    output_info_[fused_node->Name()].push_back(output_binding_indexes);
+    output_info_[fused_node->Name()].push_back(output_dim_sizes);
+    output_shapes_[fused_node->Name()] = output_shapes;
+
+    // Create function state
+    NodeComputeInfo compute_info;
+    compute_info.create_state_func = [=](ComputeContext* context, FunctionState* state) {
+      std::unique_ptr<TensorrtFuncState> p = std::make_unique<TensorrtFuncState>();
+      *p = {context->allocate_func, context->release_func, context->allocator_handle, parsers_[context->node_name].get(), engines_[context->node_name].get(), contexts_[context->node_name].get(),
+            input_info_[context->node_name], output_info_[context->node_name], output_shapes_[context->node_name]};
+      *state = p.release();
+      return 0;
+    };
+
+    // Release function state
+    compute_info.release_state_func = [](FunctionState state) {
+      if (state)
+        delete static_cast<TensorrtFuncState*>(state);
+    };
+
+    // Create compute function
+    compute_info.compute_func = [](FunctionState state, ONNXRunTimeTensor* input_tensors, size_t num_inputs, ONNXRunTimeTensor* output_tensors, size_t num_outputs) {
+      ORT_UNUSED_PARAMETER(num_inputs);
+      ORT_UNUSED_PARAMETER(num_outputs);
+      TensorrtFuncState* trt_state = reinterpret_cast<TensorrtFuncState*>(state);
+      const std::vector<int>& input_indexes = (trt_state->input_info)[0];
+      const std::vector<int>& input_binding_indexes = (trt_state->input_info)[1];
+      const std::vector<int>& input_dim_sizes = (trt_state->input_info)[2];
+      const std::vector<int>& output_binding_indexes = (trt_state->output_info)[0];
+      const std::vector<int>& output_dim_sizes = (trt_state->output_info)[1];
+      std::vector<std::vector<int64_t>> output_shapes = trt_state->output_shapes;
+      int num_binding_inputs = input_binding_indexes.size();
+      int num_binding_outputs = output_binding_indexes.size();
+      cudaStream_t stream;
+      CHECK_CUDA(cudaStreamCreate(&stream));
+      std::vector<void*> buffers(num_binding_inputs + num_binding_outputs);
+      int batch_size = 1;
+
+      // Get batch size and allocate cuda memory for inputs
+      for (int i = 0, end = num_binding_inputs; i < end; ++i) {
+        const auto& tensor_input = input_tensors[input_indexes[i]];
+        const auto& tensor_shape = tensor_input.shape;
+        const int input_batch_size = tensor_shape[0];
+        if (input_batch_size > batch_size) {
+          batch_size = input_batch_size;
+        }
+
```

Comment:
I don't think you should allow mixed batch sizes.

Reply:
Throw error if input batch size is inconsistent.